src/main/java/com/google/cloud/tools/app/impl/cloudsdk/internal/sdk/PathResolver.java

```diff
@@ -25,39 +29,70 @@
   INSTANCE;
 
   /**
-   * Attempts to find the path to Google Cloud SDK home directory.
+   * Attempts to find the path to Google Cloud SDK.
    *
    * @return Path to Google Cloud SDK or null
    */
   public Path getCloudSdkPath() {
-    String sdkDir = System.getenv("GOOGLE_CLOUD_SDK_HOME");
-    if (sdkDir == null) {
-      boolean isWindows = System.getProperty("os.name").contains("Windows");
-      if (isWindows) {
-        String programFiles = System.getenv("ProgramFiles");
-        if (programFiles == null) {
-          programFiles = System.getenv("ProgramFiles(x86)");
-        }
-        if (programFiles != null) {
-          sdkDir = programFiles + "\\Google\\Cloud SDK\\google-cloud-sdk";
+    List<String> possiblePaths = new ArrayList<>();
+
+    // search system environment PATH
+    getLocationsFromPath(possiblePaths);
+
+    // try environment variable GOOGLE_CLOUD_SDK_HOME
+    possiblePaths.add(System.getenv("GOOGLE_CLOUD_SDK_HOME"));
+
+    // search program files
+    if (System.getProperty("os.name").contains("Windows")) {
+      possiblePaths.add(getProgramFilesLocation());
+    } else {
+      // home dir
+      possiblePaths.add(System.getProperty("user.home") + "/google-cloud-sdk");
+      // try devshell VM
+      possiblePaths.add("/google/google-cloud-sdk");
+      // try bitnami Jenkins VM:
+      possiblePaths.add("/usr/local/share/google/google-cloud-sdk");
+    }
+
+    return searchPaths(possiblePaths);
+  }
+
+  private void getLocationsFromPath(List<String> possiblePaths) {
+    String pathEnv = System.getenv("PATH");
+    if (pathEnv != null) {
+      for (String path : pathEnv.split(File.pathSeparator)) {
+        // strip out trailing path separator
+        if (path.endsWith(File.separator)) {
+          path = path.substring(0, path.length() - 1);
         }
-      } else {
-        sdkDir = System.getProperty("user.home") + "/google-cloud-sdk";
-        if (!new File(sdkDir).exists()) {
-          // try devshell VM:
-          sdkDir = "/google/google-cloud-sdk";
-          if (!new File(sdkDir).exists()) {
-            // try bitnami Jenkins VM:
-            sdkDir = "/usr/local/share/google/google-cloud-sdk";
-          }
+        if (path.endsWith("google-cloud-sdk" + File.separator + "bin")) {
+          possiblePaths.add(path.substring(0, path.length() - 4));
```

Comment:
According to javadocs, it's a single characters. We don't need the trailing /. So, this should work.